### DISCUSSION TITLE
Type mismatch error ( but all correct? )

### DISCUSSION THREAD

**Discussion by SwenenzY** [prisma/prisma#22625] [Q&A] [answered]
### Question

![image](https://github.com/prisma/prisma/assets/44095096/a0dfcf10-234a-49ba-aae7-b30181450c76)

![image](https://github.com/prisma/prisma/assets/44095096/fa4ce2bc-1e16-4686-8e07-f0835e602a01)

![image](https://github.com/prisma/prisma/assets/44095096/6de04704-babd-445e-a67f-2242a3e152a8)

i want to get market_item_id : 1 and update but somehow its not working as a type error. but in types there is multiple.

    market_id?: IntFilter<"at_market_steam_items"> | number

i can use intfilter or number both are not working 

all deps are updated.

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
model at_market_steam_items {
  market_id                   Int
  steam_item_id               Int             @id @default(autoincrement())
  steam_profile_id            String?         @default("") @db.VarChar(64)
  steam_community_ban         Boolean?        @default(false)
  steam_trade_ban             Boolean?        @default(false)
  steam_is_locked             Boolean?        @default(false)
  steam_is_limited            Int?            @default(0)
  steam_level                 Int?            @default(0)
  steam_friend_count          Int?            @default(0)
  steam_country_code          String?         @default("USA") @db.VarChar(64)
  steam_currency_code         String?         @default("USD") @db.VarChar(64)
  steam_balance               Decimal?        @default(0.00) @db.Decimal(10, 5)
  steam_points                Int?            @default(0)
  steam_last_activity         Int?            @default(0)
  steam_register_date         Int?            @default(0)
  steam_hours_played_recently Decimal?        @default(0.00) @db.Decimal(10, 5)
  steam_license_count         Int?            @default(0)
  steam_packages_count        Int?            @default(0)
  steam_games_count           Int?            @default(0)
  steam_vacs                  Int[]
  steam_vac_details           Json?
  steam_profile_items         Json?
  steam_profile_games         Json?
  at_market_items             at_market_items @relation(fields: [market_id], references: [item_id], onDelete: NoAction, onUpdate: NoAction)
}
```

![image](https://github.com/prisma/prisma/assets/44095096/431ead27-07b3-4c4a-b4b1-39334d368de5)

![image](https://github.com/prisma/prisma/assets/44095096/02f11b0b-64b0-41b1-86f1-baf099044128)

**Comment by SwenenzY** [comment]
wait i can use this one accepting
```
    where: {
                steam_item_id: 1,
                market_id: 1
            },
```
but i cant use like this?
`    where: {
                market_id: 1
            },
`

why steam_item_id is required? in STANDART SQL i can do like this?
```
UPDATE at_market_steam_items
SET steam_level = 12
WHERE market_id = 1;
```

**Comment by SwenenzY** [comment]
![image](https://github.com/prisma/prisma/assets/44095096/3838fa08-ef5d-4f84-8b7c-b5e807d5b3ce)

**Comment by SwenenzY** [comment]
lol if i use updateMany everything will be solved but idk thats effecting performance cuz we are working as a shards and anti sync system.

Question : can i solve .update problem or in .updateMany performance is still same? ( 1 row update always )

**Comment by ludralph** [comment, marked as answer]
Hi @SwenenzY 👋 

Thank you for raising this question.

In Prisma, when performing an `update` operation, you must specify at least one unique field in the `where` clause. This is to ensure that the `update` operation targets a single, unique record. If `market_id` is not a unique field in your `at_market_steam_items` model, you cannot use it alone in the `where` clause for a unique update operation.
If you want to update a record based on a non-unique field, you would typically use the updateMany method. If you need to update a single record and `market_id` is not unique, you must include another unique field in the where clause, such as `steam_item_id`, which is marked with `@id` and therefore unique. See this section of the [documentation](https://www.prisma.io/docs/orm/reference/prisma-client-reference#filter-on-non-unique-fields-with-userwhereuniqueinput).

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by SwenenzY** [reply]
thanks....
